cpp/CMakeLists.txt

```diff
@@ -257,10 +282,15 @@ if(HAVE_O2_OPTIMISATION)
   list(APPEND DOLFINX_CXX_DEVELOPER_FLAGS -O2)
 endif()
 
-# Turn off some checks in gcc12 and gcc13 due to false positives with the fmt library
-if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "11.4"
-  AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "14.0")
-  list(APPEND DOLFINX_CXX_DEVELOPER_FLAGS -Wno-array-bounds;-Wno-stringop-overflow)
+# Turn off some checks in gcc12 and gcc13 due to false positives with the fmt
+# library
+if(CMAKE_COMPILER_IS_GNUCXX
```

Comment:
Does this include clang? clang sometimes identifies as GNU.

Reply:
Don't know - @chrisrichardson ? This line probably needs checking out from main anyway as it's not chsnged by my branch.

Reply:
This is deprecated in any case:

https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILER_IS_GNUCXX.html

So will switch it out for something more modern.